packages/overlay/stories/overlay-directive.stories.ts

```diff
@@ -0,0 +1,223 @@
+import '@spectrum-web-components/accordion/sp-accordion.js';
+import '@spectrum-web-components/accordion/sp-accordion-item.js';
+import '../../../projects/story-decorator/src/types.js';
+
+import './overlay-story-components.js';
+import { tooltip } from '@spectrum-web-components/tooltip/src/tooltip-directive.js';
+
+const storyStyles = html`
+    <style>
+        html,
+        body,
+        #root,
+        #root-inner,
+        sp-story-decorator {
+            height: 100%;
+            margin: 0;
+        }
+
+        sp-story-decorator::part(container) {
+            display: flex;
+            flex-direction: column;
+            width: 100%;
+            height: 100%;
+            align-items: center;
+            justify-content: center;
+        }
+
+        overlay-trigger {
+            flex: none;
+        }
+
+        #styled-div {
+            background-color: var(--styled-div-background-color, blue);
+            color: white;
+            padding: 4px 10px;
+            margin-bottom: 10px;
+        }
+
+        #inner-trigger {
+            display: inline-block;
+        }
+    </style>
+`;
+
+export default {
+    title: 'Overlay Directive',
+    argTypes: {
+        offset: { control: 'number' },
+        placement: {
+            control: {
+                type: 'inline-radio',
+                options: [
+                    'top',
+                    'top-start',
+                    'top-end',
+                    'bottom',
+                    'bottom-start',
+                    'bottom-end',
+                    'left',
+                    'left-start',
+                    'left-end',
+                    'right',
+                    'right-start',
+                    'right-end',
+                    'auto',
+                    'auto-start',
+                    'auto-end',
+                    'none',
+                ],
+            },
+        },
+        type: {
+            control: {
+                type: 'inline-radio',
+                options: ['modal', 'replace', 'inline'],
+            },
+        },
+        colorStop: {
+            control: {
+                type: 'inline-radio',
+                options: ['light', 'dark'],
+            },
+        },
+    },
+    args: {
+        placement: 'bottom',
+        offset: 0,
+        colorStop: 'light',
+        triggerOn: 'click',
+    },
+};
+
+interface Properties {
+    placement: Placement;
+    offset: number;
+    triggerOn?: OverlayContentTypes;
+    type?: Extract<TriggerInteractions, 'inline' | 'modal' | 'replace'>;
+}
+
+const template = ({
+    placement,
+    offset,
+    triggerOn,
+}: Properties): TemplateResult => {
+    return html`
+        ${storyStyles}
+        <sp-button
+            variant="primary"
+            ${tooltip(
+                () =>
+                    html`
+                        Click to open a popover.
+                    `
+            )}
+            ${trigger(
+                () => html`
```

Comment:
I know it's just a story, but I wonder what you think of naming the various parts as render functions just for readability:

```
const renderPopover = () => html`
  ...
`;

<sp-button ${trigger(renderPopover)}>
```